### PR TITLE
Function minmax v2 periods / rank / handicap (notification.handicap) / handicap (fakerank) 

### PR DESCRIPTION
**Pull request by wonderingabout**
on top of #205 

tested and ready to merge now

### CODE REVIEW

File: connection.js [hunk 1]
```diff
@@ -316,70 +316,20 @@ class Connection {
...
-            return minmaxRankFamilyReject("maxrank")
-        }
-        if ((user.ranking > config.maxrankranked) && notification.ranked) {
-            return minmaxRankFamilyReject("maxrankranked");
-        }
-        if ((user.ranking > config.maxrankunranked) && !notification.ranked) {
-            return minmaxRankFamilyReject("maxrankunranked");
+        const resultRank = genericMinmaxFamiliesReject("rank", user.ranking, notification.time_control.speed, notification.ranked, notification.handicap, user.username);
+        if (resultRank) {
+            return (resultRank);
```
Comment (Dorus) [inline connection.js]: I'm probably at fault for introducing this style, but no need for those brackets here. `return resultRank;` will work too.

File: config.js [hunk 2]
```diff
@@ -592,5 +562,5 @@ exports.updateFromArgv = function() {
             process.exit();
         }
     }
-
+
```
Comment (roy7) [inline config.js]: Bit of stray whitespace? :)

File: connection.js [hunk 3]
```diff
@@ -571,653 +492,172 @@ class Connection {
...
+        function UHMAEAT(mainPeriodTime) {
+            for (let blitzLiveCorr of ["blitz", "live", "corr"]) {
+                if (t.speed === convertBlitzLiveCorr(blitzLiveCorr)) {
+                    const minFamilyArray = familyArrayMIBLIsminIsmaxFromGeneralArgString("min" + mainPeriodTime + blitzLiveCorr);
+                    const maxFamilyArray = familyArrayMIBLIsminIsmaxFromGeneralArgString("max" + mainPeriodTime + blitzLiveCorr);
+                    // example : ["min(main/period)timeblitz", "min(main/period)timeblitzranked", "min(main/period)timeblitzunranked", 
+                    //            ["minimum", "increase", "below", "low"], [true, false]]
+
+                    minmaxRejectTests(minFamilyArray);
+                    minmaxRejectTests(maxFamilyArray);
```
Comment (Dorus) [inline connection.js]: Shouldnt the values retuned from these functions be returned somehow? Possibly only if they are not `null`?

Small Js trick that could work is `return minmaxRejectTests(minFamilyArray) || minmaxRejectTests(maxFamilyArray);`

File: connection.js [hunk 4]
```diff
@@ -571,653 +492,172 @@ class Connection {
...
+                }
+            }
+
+            function minmaxRejectTests(familyArray) {
```
Comment (Dorus) [inline connection.js]: Consider replacing `familyArray` with `[all, ranked, unranked]` and then replace `familyArray[0]` with `all`, `familyArray[1]` with `ranked` and `familyArray[2]` with `unranked`.

```
function minmaxRejectTests([all, unranked, ranked]) {
  if (config[all] || config[unranked] || config[ranked]) {
```
Another option is
```
function minmaxRejectTests(familyArray) {
  if (familyArray.some(fam => config[fam])) {
```

**Comment by wonderingabout** [comment]
changes to config.js have been moved separately to #208

**Comment by wonderingabout** [comment]
it seems current fail comes from generic minmax function, because this change (though incomplete and still mistaken) makes the test pass

https://github.com/online-go/gtp2ogs/pull/207/commits/53b39e9366042f855974f5f40a3a349785e15d8d

**Comment by wonderingabout** [comment]
actually the more i look at it the more it seems needlessly complicated, so i'll just create 2 different functions 
1) minmaxperiods
2) minmaxrank, minmaxhandicap, minmaxhandicap(fakerank)

**Comment by wonderingabout** [comment]
the mistake was small, but now i want to separate it in 2 other functions, seems wiser

**Comment by wonderingabout** [comment]
finished adding many tweaks and fixes
these simply the reject algorithm (quick exit if we wont reject), remove repetitions (mistakes easy to spot and fix), and ads tweaks and fixes here and there

**Comment by wonderingabout** [comment]
currently testing, most cases work, but currently i'm having an issue with the if all or ranked or unranked pattern :

test h0

**Comment by wonderingabout** [comment]
fixed typo in test : minhandicap = 2, maxhandicap = 5

File: connection.js (hunk 1 above)
Comment (wonderingabout) [reply connection.js]: maybe you're right !
but since i dont know about this i cant tell what should be done except from the safe simple way

**Comment by wonderingabout** [comment]
reminder for myself : for canadian periodtime test if `argNumber * notificationT.stones_per_period` or `notificationT.period_time` (which is already for X stones)

**Comment by wonderingabout** [comment]
tested and should be ready to merge now :)
testi3 fakerank 6d
test i3
testi2
testi1
test h11

**Pull request merged**